CCPA/Data Deletion Request Handling.md

```diff
@@ -0,0 +1,330 @@
+       {
+        var i = json.__uspapiCall;
+        window.__uspapi(i.command, i.version, function (retValue, success) 
+        {
+         var returnMsg = {
+         '__uspapiReturn': {
+          'returnValue': retValue,
+          'success'    : success,
+          'callId'     : i.callId
+          }
+         };
+         event.source.postMessage(msgIsString ? JSON.stringify(returnMsg) : returnMsg, '*');
+        }, i.parameter);
+       }
+      }
+      
+      uspapi_addFrame();
+      if(!('__uspapi' in window) || 
+         (typeof (window['__uspapi']) !== 'function' && 
+          typeof (window['__uspapi']) !== 'object' && 
+          (typeof (window['__uspapi']) === 'undefined' || 
+           window['__uspapi'] !== null)))
+      {
+       window['__uspapi'] = uspapi_stub;
+       window['__uspapi'].msgHandler = uspapi_msghandler;
+       if(window.addEventListener)
+       {
+        window.addEventListener('message', uspapi_msghandler, false);
+       }
+       else
+       {
+        window.attachEvent('onmessage', uspapi_msghandler);
+       }
+      }
+    </script>
+    <!-- include the vendor deletion scripts into the page: -->
+    <script src="https://vendorx.com/ccpa-delete-function.js"></script>
+  </head>
+  <body>
+    <button id="ccpa-delete">Delete my personal data</button>
+    <script>
+      document.getElementById('ccpa-delete').addEventListener('click', function ()
+      {
+       __uspapi('performDeletion', 1, null);
+      }, false);
+    </script>
+  </body>
+</html>
+```
+
+Vendor script (at `https://vendorx.com/ccpa-delete-function.js`)
+
+```
+// find the __uspapi frame
+if(!('__uspapi' in window)){
+	var f = window;
+	var cmpFrame;
+	var cmpCallbacks = {};
+	while (!cmpFrame) {
+	  try {
+		if (f.frames['__uspapiLocator']) {
+		  cmpFrame = f;
+		}
+	  } catch (e) {}
+	  if (f === window.top) {
+		break;
+	  }
+	  f = f.parent;
+	}
+
+	/* Set up a __uspapi function to do the postMessage and
+	stash the callback.
+	This function behaves (from the caller's perspective)
+	identically to the in-frame __uspapi call */
+	window.__uspapi = function(cmd, ver, callback, param) {
+	  if (!cmpFrame) {
+		callback({
+		  msg: '__uspapi not found'
+		}, false);
+		return;
+	  }
+
+	  var callId = Math.random() + '';
+	  var msg = {
+		__uspapiCall: {
+		  command: cmd,
+		  parameter: param,
+		  version: ver,
+		  callId: callId
+		}
+	  };
+	  cmpCallbacks[callId] = callback;
+	  cmpFrame.postMessage(msg, '*');
+	};
+
+	window.addEventListener('message', function(event) {
+	  var json = typeof event.data === 'string' ? JSON.parse(event.data) : event.data;
+	  if (json.__uspapiReturn) {
+		var i = json.__uspapiReturn;
+		cmpCallbacks[i.callId](i.returnValue, i.success);
+		delete cmpCallbacks[i.callId];
+	  }
+	}, false);
+
+
+	function vendorXDeletion() {
+	  //... do some deletion work …
+	  return;
+	}
+
+	__uspapi('registerDeletion', 1, vendorXDeletion);
```

Comment:
I think this call to `__uspapi` needs to move outside the closing bracket on line 328 right @janwinkler?